### ISSUE TITLE
CollectionModified exception in new drag cube system

### ISSUE THREAD

**Issue by JonnyOThan** [KSPModdingLibs/KSPCommunityFixes#146]
noticed this when looking at someone else’s logs. Not sure if it’s serious but figured I’d open this for investigation.

```
[EXC 20:24:42.578] InvalidOperationException: Collection was modified; enumeration operation may not execute.
	System.ThrowHelper.ThrowInvalidOperationException (System.ExceptionResource resource) (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Collections.Generic.List`1+Enumerator[T].MoveNextRare () (at <9577ac7a62ef43179789031239ba8798>:0)
	System.Collections.Generic.List`1+Enumerator[T].MoveNext () (at <9577ac7a62ef43179789031239ba8798>:0)
	KSPCommunityFixes.Performance.DragCubeGeneration+<RenderDragCubesOnCopy>d__21.MoveNext () (at <0d4923dd57ad48cfb66dc63094990ea9>:0)
```
Full log here: https://discord.com/channels/542919794081660929/931305615228305428/1105561653573062748

(sorry for the discord link but I’m on my phone. I’ll try and get the file uploaded directly later)

**Comment by gotmachine** [comment]
Thanks for forwarding, however the discord link doesn't seem to be valid, and that would really help to know at which point this happen exactly.

**Comment by JonnyOThan** [comment]
Here's the full log.
[Uploading KSP (20).log…]()

**Comment by gotmachine** [comment]
Should be fixed in 1.28.1
